rxpager/src/test/java/com/schibstedspain/android/rxpager/PagerTest.java

```diff
@@ -34,131 +36,125 @@
   }
 
   @Test
-  public void getPageObservableShouldReturnJustOnePageIfThereIsOnlyOne() {
+  public void getPageObservableShouldReturnJustOnePageIfThereIsOnlyOne() throws Exception {
     givenThereIsOnePage();
 
-    TokenPage<String> firstPage = pager.getPageObservable()
-        .toBlocking()
-        .single();
+    TokenPage<String> firstPage = pager.getPageObservable().blockingFirst();
 
-    verify(getPageMock).call(anyString());
+    verify(getPageMock).apply(anyString());
     assertEquals(SINGLE_PAGE, firstPage);
   }
 
   @Test
-  public void getPageObservableShouldReturnJustTheFirstPageEvenIfThereAreMore() {
+  public void getPageObservableShouldReturnJustTheFirstPageEvenIfThereAreMore() throws Exception {
     givenThereAreThreePages();
 
-    TokenPage<String> page = pager.getPageObservable().toBlocking().first();
+    TokenPage<String> page = pager.getPageObservable().blockingFirst();
 
-    verify(getPageMock).call(anyString());
+    verify(getPageMock).apply(anyString());
     assertEquals(FIRST_PAGE, page);
   }
 
   @Test
-  public void hasMoreShouldReturnFalseWhenThereAreNoMore() {
+  public void hasMoreShouldReturnFalseWhenThereAreNoMore() throws Exception {
     givenThereIsOnePage();
-    pager.getPageObservable()
-        .toBlocking()
-        .single();
+    pager.getPageObservable().blockingFirst();
 
     boolean hasNext = pager.hasNext();
 
     assertFalse(hasNext);
   }
 
   @Test
-  public void hasMoreShouldReturnTrueWhenThereAreMore() {
+  public void hasMoreShouldReturnTrueWhenThereAreMore() throws Exception {
     givenThereAreThreePages();
-    pager.getPageObservable()
-        .toBlocking()
-        .first();
+    pager.getPageObservable().blockingFirst();
 
     boolean hasNext = pager.hasNext();
 
     assertTrue(hasNext);
   }
 
   @Test
-  public void nextShouldGiveTheSecondPage() {
+  public void nextShouldGiveTheSecondPage() throws Exception {
     givenThereAreThreePages();
-    TestSubscriber<TokenPage<String>> testSubscriber = new TestSubscriber<>();
-    waitUntilLoaded(() -> pager.getPageObservable().subscribe(testSubscriber));
+    TestObserver<TokenPage<String>> testObserver = new TestObserver<>();
 
+    waitUntilLoaded(() -> pager.getPageObservable().subscribe(testObserver));
     waitUntilLoaded(() -> pager.next());
 
-    verify(getPageMock, times(2)).call(anyString());
-    testSubscriber.assertReceivedOnNext(Arrays.asList(FIRST_PAGE, SECOND_PAGE));
+    verify(getPageMock, times(2)).apply(anyString());
+    testObserver.assertValueSequence(Arrays.asList(FIRST_PAGE, SECOND_PAGE));
   }
 
-  private void waitUntilLoaded(Action0 action) {
-    TestSubscriber<Boolean> loadingSubscriber = new TestSubscriber<>();
+  private void waitUntilLoaded(Action action) throws Exception {
+    TestObserver<Boolean> loadingSubscriber = new TestObserver<>();
     pager.getIsLoadingObservable().take(3).subscribe(loadingSubscriber);
```

Comment:
Non-blocking comment: You could use the `.test()` instead of creating a `TestObserver`.

Reply:
done